lib/errors.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+class TldrError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = 'TldrError';
+  }
+
+  static isTldrError(err) {
+    return err instanceof this;
+  }
+}
+
+class EmptyCacheError extends TldrError {
+  constructor() {
+    super(trim`
+Local cache is empty
+Please run tldr --update
+    `);
+    this.name = 'EmptyCacheError';
+    // eslint-disable-next-line no-magic-numbers
+    this.code = 2;
+  }
+}
+
+class MissingPageError extends TldrError {
+  constructor(repo) {
+    super(trim`
+Page not found.
+If you want to contribute it, feel free to send a pull request to: ${repo}
+    `);
+    this.name = 'MissingPageErrror';
+    // eslint-disable-next-line no-magic-numbers
+    this.code = 3;
+  }
+}
+
+module.exports = {
+  TldrError,
+  EmptyCacheError,
+  MissingPageError
+};
+
+function trim(strings, ...values) {
```

Comment:
What is the purpose of this? Is it just to remove the newlines in the original string?

Reply:
Syntactical sugar to replace the following form:
```js
    super(`
Page not found.
If you want to contribute it, feel free to send a pull request to: ${repo}
    `.trim());
```
with slightly more readable IMHO:
```js
    super(trim`
Page not found.
If you want to contribute it, feel free to send a pull request to: ${repo}
    `);
```